apps/events/models.py

```diff
@@ -0,0 +1,165 @@
+from datetime import timedelta
+
+from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.db.models import F, Func, OuterRef, Q, Subquery
+from django.utils import timezone
+
+from apps.chat.models import ChatMessage, ChatMessageType
+from apps.events.actions import end_conversation, log, summarize_conversation
+from apps.experiments.models import Experiment, ExperimentSession
+from apps.utils.models import BaseModel
+
+ACTION_FUNCTIONS = {"log": log, "end_conversation": end_conversation, "summarize": summarize_conversation}
+
+
+class EventActionType(models.TextChoices):
+    LOG = "log"  # Prints the last message
+    END_CONVERSATION = "end_conversation"  # Ends the conversation
+    SUMMARIZE = "summarize"  # Summarize the conversation
+
+
+class EventAction(BaseModel):
+    action_type = models.CharField(choices=EventActionType.choices)
```

Comment:
I'm torn about putting the choices in here. 
`<rant>`
It really irritates me that any time you change the Django makes a new migration file.
`</rant>`